File: 0007 Reverse Integer/ReverseInteger.go
package main

import (
	"fmt"
	"math"
)

func main() {
	fmt.Println(reverse(123))
}

var max = math.MaxInt32 / 10
var min = math.MinInt32 / 10

func reverse(x int) int {
	ret := 0

	for x != 0 {
		digit := x % 10

		if ret > max || (ret == max && digit > 7) {
			return 0
		}
		if ret < min || (ret == min && digit < -8) {
			return 0
		}

		ret = ret*10 + digit
		x /= 10
	}
	return ret
}

File: 0007 Reverse Integer/ReverseInteger.c
const int max = INT_MAX / 10;
const int min = INT_MIN / 10;

int reverse(int x){
    int ret = 0;
    
    while (x != 0) {
        int digit = x % 10;

        if (ret > max || (ret == max && digit > 7)) {
			return 0;
		}
		if (ret < min || (ret == min && digit < -8)) {
			return 0;
		}

		ret = ret * 10 + digit;
		x /= 10;
    }
    return ret;
}